src/lib/retry.sh

```diff
@@ -8,31 +8,27 @@ set -euo pipefail
 #
 # Successive backoffs double the timeout.
 function retry {
+  local n=1
   local max_attempts=${ATTEMPTS:-5}
   local timeout=${TIMEOUT:-1}
   local attempt=1
   local exitCode=0
 
-  while (( attempt <= max_attempts ))
-  do
-    if "$@"
-    then
-      return 0
-    fi
-    exitCode=$?
-
-    >&2 echo "Attempt #$attempt/$max_attempts failed."
-    attempt=$(( attempt + 1 ))
-    (( attempt > max_attempts )) && break
-
-    echo "Retrying in $timeout seconds ..."
-    sleep "$timeout"
-    timeout=$(( timeout * 2 ))
+  while true; do
+    "$@" && break || {
+      exitCode=$?
+      echo "the exitcode is $exitCode"
+      if [[ $n -lt $max_attempts ]]; then
+        ((n++))
+        echo "Command failed. Attempt $n/$max_attempts. Retrying after $timeout seconds."
```

Comment:
n is initialised as 1, but is incremented before being printed. The first number it will display is 2/5